modules/store/src/utils.ts

```diff
@@ -92,10 +93,17 @@ export function createReducerFactory<T, V extends Action = Action>(
   metaReducers?: MetaReducer<T, V>[]
 ): ActionReducerFactory<T, V> {
   if (Array.isArray(metaReducers) && metaReducers.length > 0) {
-    return compose.apply(null, [...metaReducers, reducerFactory]);
+    reducerFactory = compose.apply(null, [...metaReducers, reducerFactory]);
   }
 
-  return reducerFactory;
+  return (rm: ActionReducerMap<T, V>, initialState?: InitialState<T>) => {
+    const reducer = reducerFactory(rm, initialState);
+
+    return (state: T | undefined, action: V) => {
+      state = state === undefined ? (initialState as T) : state;
+      return typeof reducer === 'function' ? reducer(state, action) : state;
```

Comment:
@timdeschryver 
I think it's a valid point, but keep in mind that the `reducerFactory` can be provided by the user.
anyway let me know what you have decided :)

Reply:
Let's leave it like this.

Reply:
Sorry, I was to fast... leaving it as is, is the safer option, but I personally would want to remove it. Redux is more or less just the `combineReducers` function, which should return a reducer function. Since this is a hot code path (called multiple times with each action dispatch), I believe we should remove the check. If there are users providing their own factory function and are experiencing issues we can put in the check.

Reply:
Ok, so I will remove the check and update the failing test :)